examples/bmi-for-nextgen.ipynb

```diff
@@ -0,0 +1,1696 @@
+    "\n",
+    "**Modifying**\n",
+    "\n",
+    "2) There is no way to change parameter values, file paths, etc. without changing the `.py` model file itself. What if you or a colleague want to deploy the model in a different location or over a different temporal extent and time step?\n",
+    "\n",
+    "3) The only way to add new processes/capabilities to the model is to change the single file in the source code. There are no modules that can be swapped in and out. You can't call functions from a central driver or program. You cannot run unit tests (because there are no functional units to test).\n",
+    "\n",
+    "**Coupling**\n",
+    "\n",
+    "4) There are only two ways to couple this model: through soft coupling (running the model, saving its output, and using that as forcing for another model) or changing its source code entirely.\n",
+    "\n",
+    "**Controlling**\n",
+    "\n",
+    "5) A centralized framework (e.g., NextGen) cannot control model runtime in any standardized way."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b8b986e",
+   "metadata": {},
+   "source": [
+    "# 2 The right way\n",
+    "\n",
+    "Fortunately, we can remedy most of these issues by first making `snowPy` BMI compliant and then implementing the full set of BMI functions.\n",
+    "\n",
+    "## 2.1 BMI background\n",
+    "\n",
+    "The Community Surface Dynamics Modeling System ([CSDMS](https://csdms.colorado.edu/)) group at the University of Colorado Boulder develops and maintains BMI. You can find further information on the [BMI GitHub page](https://github.com/csdms/bmi) or on the [CSDMS Wiki](https://csdms.colorado.edu/wiki/BMI).\n",
+    "\n",
+    "The two reference papers for BMI are:\n",
+    "\n",
+    "- *Hutton, E.W., Piper, M.D., and Tucker, G.E., 2020. The Basic Model Interface 2.0: A standard interface for coupling numerical models in the geosciences. Journal of Open Source Software, 5(51), 2317, https://doi.org/10.21105/joss.02317.*\n",
+    "- *Peckham, S.D., Hutton, E.W., and Norris, B., 2013. A component-based approach to integrated modeling in the geosciences: The design of CSDMS. Computers & Geosciences, 53, pp.3-12, http://dx.doi.org/10.1016/j.cageo.2012.04.002.*\n",
+    "\n",
+    "The CSDMS `heat` [example](https://github.com/csdms/bmi-example-python) is a great reference for BMI implementations. We based the `snowBMI` model, which we'll see shortly, heavily on the CSDMS examples. \n",
+    "\n",
+    "## 2.2 BMI philosophy\n",
+    "\n",
+    "BMI is middleware, meaning it is software that operates between other components (in our case a centralized framework—NextGen—and a model). It standardizes model querying, control, and coupling. \n",
+    "\n",
+    "BMI works because the functions are the same across models and languages. Making a model BMI compliant can remove (or mask) model idiosyncrasies.\n",
```

Comment:
That's a pretty strong statement that is demonstrably false.  I would consider "BMI woks because a similar set of functions with defined semantics are consistently applied across models and languages." Or something along those lines.

Reply:
Is there a version of what you're wanting to say that is interpretable by people without CS degrees? My point wasn't that the function contents are the same (obviously a fortran function != a C function, for example), but that they have the same names and produce similar, expected outcomes (get_time_step, for example, will return the model's time step across languages and models).

Reply:
Update: I just removed this line of text to reduce confusion.